porcupine/plugins/rightclick_menu.py

```diff
@@ -17,45 +17,33 @@ def text_is_selected(tab: tabs.FileTab) -> bool:
         return False
     return True
 
+
 def show_menu(event) -> None:
     rm = tkinter.Menu(get_main_window(), tearoff=0)
 
     for path, func in _menu_items.items():
         rm.add_command(label=path, command=func)
 
-
-    flag = bool
-    try:
-        flag=text_is_selected(get_filetab())
-    except AssertionError:
-        flag=False
-    finally:
-        if not flag:
-            for i in _ftab_indexes:
-                rm.entryconfigure(i, state=tkinter.DISABLED)
+    flag = text_is_selected(get_filetab())
+    if not flag:
+        for i in _ftab_indexes:
+            rm.entryconfigure(i, state=tkinter.DISABLED)
 
     rm.tk_popup(event.x_root, event.y_root)
     rm.bind("<Unmap>", (lambda event: rm.after_idle(rm.destroy)), add=True)
 
-def add_rightclick_option(path: str, func, needs_selected_text: bool = False):
+
+def add_rightclick_option(path: str, func, needs_selected_text: bool = False) -> None:
     if not needs_selected_text:
         _menu_items[path] = func
     else:
-        if func is None:
-            command = lambda: get_filetab().event_generate(f"<<FiletabCommand:{path}>>")
-        else:
-            command = lambda: func(get_filetab())  # type: ignore
-
-        _menu_items[path] = command
+        _menu_items[path] = lambda: func(get_filetab())
```

Comment:
I think we should do this in both cases. The `needs_selected_text` shouldn't affect what ends up in `_menu_items`, it should only affect whether we append to `_ftab_indexes`.